catalog/nsili/catalog-nsili-endpoint/src/main/java/com/connexta/alliance/nsili/endpoint/requests/SubmitQueryRequestImpl.java

```diff
@@ -28,20 +32,57 @@
 import com.connexta.alliance.nsili.common.UCO.StringDAGListHolder;
 import com.connexta.alliance.nsili.common.UCO.SystemFault;
 import org.omg.CORBA.NO_IMPLEMENT;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Result;
 
 public class SubmitQueryRequestImpl extends SubmitQueryRequestPOA {
 
+    //TODO TROY
+    private static final String ENCODING = "UTF-8";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SubmitQueryRequestImpl.class);
+
+    private int maxNumReturnedHits = -1;
+
+    private List<Result> queryResults;
+
+    public void setQueryResults(List<Result> queryResults) {
+        this.queryResults = queryResults;
+    }
+
     @Override
     public void set_number_of_hits(int hits)
             throws ProcessingFault, InvalidInputParameter, SystemFault {
-        return;
+        this.maxNumReturnedHits = hits;
     }
 
     @Override
-    public State complete_DAG_results(DAGListHolder results)
-            throws ProcessingFault, SystemFault {
-        DAG[] result = null;  //TODO TROY DAGGenerator.generateDAGResultNSILAllView(_orb());
-        results.value = result;
+    public State complete_DAG_results(DAGListHolder results) throws ProcessingFault, SystemFault {
+        DAG[] noResults = new DAG[0];
+        if (queryResults == null) {
+            results.value = noResults;
+        } else {
+            List<DAG> dags = new ArrayList<>();
+            for (Result result : queryResults) {
+                DAG dag = ResultDAGConverter.convertResult(result, _orb(), _poa());
+                if (dag != null) {
+                    dags.add(dag);
+                }
+            }
+            if (!dags.isEmpty()) {
+                results.value = dags.toArray(new DAG[queryResults.size()]);
+
+                //TODO REMOVE
+                LOGGER.debug("Number of results being returned: "+results.value.length+ " requested: "+maxNumReturnedHits);
+            } else {
+                //TODO REMOVE
+                LOGGER.debug("{} : No results will be returned");
```

Comment:
No results are being printed here, but It would be useful to keep as a debug message.